index.html

```diff
@@ -3957,77 +4078,73 @@ <h3>Algorithm</h3>
         to be compacted.</p>
 
       <ol>
-        <li>Initialize <var>number entries</var> to the number of <a>entries</a>
-          <var>value</var> contains.</li>
-        <li>If <var>value</var> has an <code>@index</code> <a>entry</a> and the
-          <a>container mapping</a> associated to <var>active property</var>
-          <span class="changed">includes</span> <code>@index</code>, decrease <var>number entries</var> by
-          <code>1</code>.</li>
-        <li>If <var>number entries</var> is greater than <code>2</code>, return
-          <var>value</var> as it cannot be compacted.</li>
-        <li class="changed">If the <a>type mapping</a> of <var>active property</var> is <code>@none</code>,
-          leave <var>value</var> as is, as value compaction is disabled.
-          <ol>
-            <li>Replace any value of <code>@type</code> in <var>value</var> with the result of using the
-              <a href="#iri-compaction">IRI compaction algorithm</a>,
-              passing <var>active context</var>, <var>inverse context</var>,
-              the value of the <code>@type</code> <a>entry</a> for <var>var</var>, and
-              <code>true</code> for <var>vocab</var>.</li>
-            <li>Replace each key in <var>value</var> with the result of using the
-              <a href="#iri-compaction">IRI compaction algorithm</a>,
-              passing <var>active context</var>, <var>inverse context</var>,
-              the that key for <var>var</var>, and
-              <code>true</code> for <var>vocab</var>.</li>
-            <li>Return <var>value</var>.</li>
-          </ol>
-        </li>
-        <li>If <var>value</var> has an <code>@id</code> <a>entry</a>:
+        <li>Initialize <var>result</var> to <var>value</var>.</li>
+        <li>Initialize <var>language</var> to the <a>language mapping</a> for <var>active property</var>
+          in <var>active context</var>, if any, otherwise to the <a>default language</a>
+          of <var>active context</var>.</li>
+        <li class="changed">Initialize <var>direction</var> to the <a>direction mapping</a> for <var>active property</var>
+          in <var>active context</var>, if any, otherwise to the <a>default base direction</a>
+          of <var>active context</var>.</li>
+        <li>If <var>value</var> has an <code>@id</code> <a>entry</a>
+          and has no other <a>entries</a> other than `@index`:
           <ol>
-            <li>If <var>number entries</var> is <code>1</code> and
+            <li>If <var>value</var> and
               the <a>type mapping</a> of <var>active property</var>
-              is set to <code>@id</code>, return the result of using the
+              is set to <code>@id</code>, set <var>result</var> to the result of using the
```

Comment:
It could be done that way, and was previously, but I found it better to simply update _result_, to allow for key transformation, if it is still a _map_, before returning it.

Reply:
My understanding was that the IRI compaction algorithm returned a string, so there would be no more key to transform... hence my proposal to return it immediately.
Or did I miss something in the IRI compaction algorithm?

Reply:
No, you're right, and it could be returned early, but there's something to be said for having the exit at the end of the algorithm. Having returns scattered through the algorithm might be less obvious, and the "Otherwise" clauses make it clear that you get to the end, where we see if it's a map.

But, if the consensus is to do early returns, I'll go along with that. @dlongley?

Reply:
(We should probably move this to a new issue or PR, as this PR is closed, and it can be confusing what conversations were resolved in the code, and what is still open for consideration).

Reply:
@gkellogg 
> there's something to be said for having the exit at the end of the algorithm

very true

> and the "Otherwise" clauses make it clear that you get to the end

correct, I overlooked that.

So I'm happy with keeping those as 'set result'...